src/content/language/collections.md

```diff
@@ -242,78 +252,355 @@ final constantMap = const {2: 'helium', 10: 'neon', 18: 'argon'};
+var items = [
+  0,
+  if (a case 'apple') 1 else 10,
+  if (b case 'pear') 2 else 20,
+  3,
+  4,
+]; // [0, 1, 20, 3, 4]
+```
+
+<?code-excerpt "misc/lib/language_tour/misc/if_case_operator_in_collection_c.dart (code_sample)"?>
+```dart
+var a = 'apple';
+var b = 'orange';
+var c = 'mango';
+var items = [
+  0,
+  if (a case 'apple') 1 else if (a case 'mango') 10,
+  if (b case 'pear') 2 else if (b case 'mango') 20,
+  if (c case 'apple') 3 else if (c case 'mango') 30,
+  4,
+]; // [0, 1, 30, 4]
+```
+
+<?code-excerpt "misc/lib/language_tour/misc/if_case_operator_in_collection_d.dart (code_sample)"?>
+```dart
+var word = 'hello';
+var items = [
+  1,
+  if (word case String(length: var word_length)) word_length,
+  3,
+]; // [1, 5, 3]
+```
+
+To learn more about `if-case` conditionals,
+see [`if-case` statement][].
+
+[`if-case` statement]: /language/branches#if-case
+
+### For loop elements {: #for-element }
+
+You can use a `for` element inside of a collection to
+insert multiple elements into that collection.
+
+A `for` element has this syntax in a collection:
+
+```dart
+for (<expression> in <collection>) <result>
+```
+
+```dart
+for (<initialization_clause>; <condition_clause>; <increment_clause>) <result>
+```
+
+The following examples illustrate various ways that
+you can use a `for` element inside of a collection:
+
+<?code-excerpt "misc/lib/language_tour/misc/for_loop_in_collection_a.dart (code_sample)"?>
+```dart
+var numbers = [2, 3, 4];
+var items = [1, for (var n in numbers) n * n, 7]; // [1, 4, 9, 16, 7]
+```
+
+<?code-excerpt "misc/lib/language_tour/misc/for_loop_in_collection_b.dart (code_sample)"?>
+```dart
+var items = [1, for (var x = 5; x > 2; x--) x, 7]; // [1, 5, 4, 3, 7]
+```
+
+<?code-excerpt "misc/lib/language_tour/misc/for_loop_in_collection_c.dart (code_sample)"?>
+```dart
+var items = [1, for (var x = 2; x < 4; x++) x, 7]; // [1, 2, 3, 4, 7]
+```
+
+To learn more about the `for` loop, see
+[`for` loops][].
+
+[`for` loops]: /language/loops/#for-loops
+
+### Nesting elements {: #nesting-elements }
+
+You can nest `if` elements, `for` elements, and spread
+elements within each other. This is a powerful alternative
+to list comprehensions in other languages.
+
+In the following example, only the even numbers in
+`numbers` are included in `items`.
+
+<?code-excerpt "misc/lib/language_tour/misc/nesting_in_collection_a.dart (code_sample)"?>
+```dart
+var numbers = [1, 2, 3, 4, 5, 6, 7];
+var items = [
+  0,
+  for (var n in numbers)
+    if (n.isEven) n,
+  8,
+]; // [0, 2, 4, 6, 8]
+```
+
+It's common and idiomatic to use a spread on a collection
+literal immediately inside of an `if` or `for` element. For
+example:
+
+<?code-excerpt "misc/lib/language_tour/misc/nesting_in_collection_c.dart (code_sample)"?>
+```dart
+var items = [
+  if (condition) oneThing(),
+  if (condition) ...[multiple(), things()],
+]; // [oneThing, [multiple_a, multiple_b], things]
+```
+
+You can nest all kinds of elements arbitrarily deep.
```

Comment:
Is it worth clarifying besides the leaf elements (expressions and map entries)?

Reply:
Tentatively closing due to the age of the ask. Feel free to modify this section in a followup PR if you feel like it.